pypdf/_writer.py

```diff
@@ -2417,17 +2417,45 @@ def pageMode(self, mode: PagemodeType) -> None:  # deprecated
         deprecation_with_replacement("pageMode", "page_mode", "3.0.0")
         self.page_mode = mode
 
-    def add_annotation(self, page_number: int, annotation: Dict[str, Any]) -> None:
+    def add_annotation(
+        self,
+        page_number: Union[int, PageObject],
+        annotation: Dict[str, Any],
+        pagenumber: Any = None,
+    ) -> DictionaryObject:
+        """
+        Add a single annotation to the page. Must be a new annotation (can not be recycled)
+
+        Args:
+            page: page object or number (used to be pagenumber : deprecated)
+            annotation : annotation to be added (created with annotation)
+
+        Returns:
+            the inserted object (to be used in pop-up creation argument for example)
+        """
+        if pagenumber is not None:  # deprecated
+            deprecation_with_replacement(
+                "add_annotation(page_number,annotation)",
+                "add_annotation(page,annotation)",
+                "3.0.0",
+            )
+            page = pagenumber
```

Comment:
I've just checked:

* https://github.com/py-pdf/pypdf/blob/3.0.0/PyPDF2/_writer.py : `pagenumber` is not here
* https://github.com/py-pdf/pypdf/blob/2.12.1/PyPDF2/_writer.py#L2054 : `pagenumber` is also not here

Reply:
It was in progress on my computer... Some regression were to be fixed first.